fetch.js

```diff
@@ -66,21 +75,31 @@ function GetFileFromUrl(inputUrl, processFileCallback) {
         res.setEncoding("utf8");
         res.on("data", function (chunk) { rawResponse += chunk; });
         res.on("end", function () {
-            console.log("<- Finished reading: " + inputUrl);
-            processFileCallback(rawResponse);
+            console.log("<- Finished reading " + rawResponse.length + " bytes: " + inputUrl);
+            if (!processFileCallback(rawResponse))
+                console.log("  ***  Failed to GetFileFromUrl: " + inputUrl);
         });
     });
 }
 
-function GetApiFile(inputUrl, outputFilename) {
+function GetApiFile(inputUrl, outputFilename, retry = 0) {
+    if (retry == 10)
+        throw new Exception("  !!!!!!!!!!  Aborting GetApiFile: " + inputUrl);
+
     console.log("-> Begin reading: " + inputUrl);
     var rawResponse = "";
     var postReq = https.get(inputUrl, function (res) {
         res.setEncoding("utf8");
         res.on("data", function (chunk) { rawResponse += chunk; });
         res.on("end", function () {
-            console.log("<- Finished reading: " + inputUrl);
-            WriteJsonFile(outputFilename, TabifyJson(rawResponse, outputFilename, 2));
+            console.log("<- Finished reading " + rawResponse.length + " bytes: " + inputUrl);
+            var jsonOutput = TabifyJson(rawResponse, outputFilename, 2);
+            if (jsonOutput)
+                WriteJsonFile(outputFilename, jsonOutput);
+            else {
+                console.log("  !!!  Failed to GetApiFile (" + retry + "): " + inputUrl);
```

Comment:
>( [](start = 57, length = 1)

Maybe "... GetApiFile (retry " + retry + ")... " so that we won't forget what the number in the brackets means? :)